spec/builtins2.test.sh

```diff
@@ -121,6 +121,101 @@ command command -v seq
 ## N-I zsh stdout-json: ""
 ## N-I zsh status: 127
 
+#### history -a
+case $SH in dash|mksh|zsh) exit 0 ;; esac
```

Comment:
After this PR we should probably split this into  `spec/builtins-history.test.sh`

If a feature is only implemented in bash, and not other shells, then that's a good reason to have another file

I just did that with `spec/builtin-trap.test.sh` and `spec/builtin-trap-bash.test.sh`